src/arithmetic.jl

```diff
@@ -96,19 +80,12 @@ for (f, fc) in ((:+, :(add!)), (:-, :(subst!)))
             ($f){T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = $f(promote(a,b)...)
 
             function $f{T<:Number}(a::$T{T}, b::$T{T})
-                la = a.order+1
-                lb = b.order+1
-                v = similar(a.coeffs, max(la,lb))
-                if a.order == b.order
-                    @__dot__ v = $f(a.coeffs, b.coeffs)
-                elseif a.order < b.order
-                    @inbounds @__dot__ v[1:la] = $f(a[1:la], b[1:la])
-                    @inbounds @__dot__ v[la+1:lb] = $f(b[la+1:lb])
-                else
-                    @inbounds @__dot__ v[1:lb] = $f(a[1:lb], b[1:lb])
-                    @inbounds @__dot__ v[lb+1:la] = a[lb+1:la]
+                if a.order != b.order
+                    a, b = fixorder(a, b)
                 end
-                return $T(v)
+                v = similar(a.coeffs)
```

Comment:
You certainly can simply do `v = $f(a.coeffs, b.coeffs)`, but as far as I understand, using `@__dot__` avoids creating some temporary arrays and copying the results, which is therefore faster. For the type of benchmarks that Jorge was checking in #102, without `@__dot__` in the addition functions produces ~30% more memory usage, and ~10% slower code.